src/test/system_tests/ceph_s3_tests/test_ceph_s3.js

```diff
@@ -101,18 +101,18 @@ async function test_worker() {
 }
 
 async function run_single_test(test) {
-    let ceph_args = `S3TEST_CONF=${CEPH_TEST.test_dir}${CEPH_TEST.ceph_config}`;
+    let ceph_args = `S3TEST_CONF=${process.cwd()}/${CEPH_TEST.test_dir}${CEPH_TEST.ceph_config}`;
+    const tox_args = `-c ${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}${CEPH_TEST.tox_config}`;
     if (S3_CEPH_TEST_SIGV4.includes(test)) {
         ceph_args += ` S3_USE_SIGV4=true`;
     }
-    let base_cmd = `${ceph_args} ./${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}virtualenv/bin/nosetests`;
+    let base_cmd = `${ceph_args} tox ${tox_args}`;
     if (!S3_CEPH_TEST_OUT_OF_SCOPE_REGEXP.test(test)) {
         try {
-            const test_name = test.replace(S3_CEPH_TEST_STEMS_REGEXP, pref => `${pref.slice(0, -1)}:`); //Match against the common test path
-            if (test_name.includes('boto')) {
-                base_cmd = `${ceph_args} ./${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}virtualenv/bin/nosetests -v -s -A 'not fails_on_rgw'`;
+            if (test.includes('boto')) {
+                base_cmd = `${ceph_args} tox ${tox_args} -- -m 'not fails_on_aws'`;
             }
-            const res = await os_utils.exec(`${base_cmd} ${test_name}`, { ignore_rc: false, return_stdout: true });
+            const res = await os_utils.exec(`${base_cmd} ${process.cwd()}/${CEPH_TEST.test_dir}${CEPH_TEST.s3_test_dir}${test}`, { ignore_rc: false, return_stdout: true });
```

Comment:
I suggest to define a variable `test_name` as it was and then this line would be more readable.